pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -37,22 +37,32 @@ type ResolvedResultRef struct {
 }
 
 // ResolveResultRefs resolves any ResultReference that are found in the target ResolvedPipelineRunTask
-func ResolveResultRefs(pipelineRunState PipelineRunState, targets PipelineRunState) (ResolvedResultRefs, error) {
+func ResolveResultRefs(pipelineRunState PipelineRunState, targets PipelineRunState, pipelineResults []v1beta1.PipelineResult) (ResolvedResultRefs, error) {
 	var allResolvedResultRefs ResolvedResultRefs
-	for _, target := range targets {
-		resolvedResultRefs, err := convertParamsToResultRefs(pipelineRunState, target)
-		if err != nil {
-			return nil, err
+	if targets != nil {
+		for _, target := range targets {
+			resolvedResultRefs, err := convertParamsToResultRefs(pipelineRunState, target)
+			if err != nil {
+				return nil, err
+			}
+			allResolvedResultRefs = append(allResolvedResultRefs, resolvedResultRefs...)
+		}
+	}
+	if len(pipelineResults) != 0 {
+		for _, result := range pipelineResults {
+			resolvedResultRefs := convertPipelineResultToResultRefs(pipelineRunState, result)
+			if resolvedResultRefs != nil {
+				allResolvedResultRefs = append(allResolvedResultRefs, resolvedResultRefs...)
+			}
 		}
-		allResolvedResultRefs = append(allResolvedResultRefs, resolvedResultRefs...)
 	}
 	return removeDup(allResolvedResultRefs), nil
 }
 
-// extractResultRefsFromParam resolves any ResultReference that are found in param
+// extractResultRefs resolves any ResultReference that are found in param or pipeline result
 // Returns nil if none are found
-func extractResultRefsFromParam(pipelineRunState PipelineRunState, param v1beta1.Param) (ResolvedResultRefs, error) {
-	if resultRefs, err := v1beta1.NewResultRefs(param); err == nil {
+func extractResultRefs(pipelineRunState PipelineRunState, obj interface{}) (ResolvedResultRefs, error) {
```

Comment:
similarly to other comments I suggest passing a `string` to this func instead of an `interface{}`. `convertPipelineResultToResultRefs` and `convertParamsToResultRefs` can each decide where to pull that string from (`param.Value.StringVal` or `pipelineResult.Value`).

Reply:
done.